packages/did/src/IdenityRegistry.ts

```diff
@@ -59,13 +31,20 @@ export class IdenityRegistry {
    * @returns {Promise} Promise object represents the result. The result
    *   conatains DID `did` and DID document `document`.
    */
-  publish = async (
-    seed: Seed,
-    publicKeys: PublicKeyPem[] = [],
-  ) => {
-    const network = '0x1';
-    const ticClient = await this.getTicClient(network, seed);
-    const did = encodeToDid({ network, address: ticClient.masterRoot });
+  public async publish(seed: Seed, publicKeys: PublicKeyPem[] = []) {
+    const iota = mamClient.getIota(this.provider);
+    const client = await mamClient.createMamFrom({
+      iota,
+      seed,
+      mode: 'public',
+    });
+
+    if (client.mam.channel.start !== 0) {
```

Comment:
You shall append related comments about the whole process.

Reply:
Added more comment about the whole process.